crates/sui-source-validation/src/tests.rs

```diff
@@ -264,29 +264,33 @@ async fn package_not_found() -> anyhow::Result<()> {
         panic!("Expected verification to fail");
     };
 
-    let expected = expect!["Dependency object does not exist or was deleted: ObjectNotFound { object_id: 0x<b_id>, version: None }"];
+    let expected = expect!["Dependency object does not exist or was deleted: ObjectNotFound { object_id: 0x<id>, version: None }"];
     expected.assert_eq(&sanitize_id(err.to_string(), &stable_addrs));
 
+    let package_root = AccountAddress::random();
+    stable_addrs.insert(SuiAddress::from(package_root), "<id>");
     let Err(err) = verifier.verify_package_root_and_deps(
 	&a_pkg.package,
-	/* Subst address here doesnt matter */ AccountAddress::random(),
+	package_root,
     ).await else {
 	panic!("Expected verification to fail");
     };
 
-    let expected = expect!["Dependency object does not exist or was deleted: ObjectNotFound { object_id: 0x<b_id>, version: None }"];
+    // <id> below may refer to either the package_root or dependent package `b`
+    // (the check reports the first missing object nondeterministically)
+    let expected = expect!["Dependency object does not exist or was deleted: ObjectNotFound { object_id: 0x<id>, version: None }"];
```

Comment:
The flake can happen here, where the expected value should just be `id`. Other tests are updated to use `id` for good measure.